presto-main/src/test/java/com/facebook/presto/sql/planner/TestEarlyOutJoins.java

```diff
@@ -0,0 +1,247 @@
+/*
```

Comment:
Please add test case with the pattern 't1.a1 IN (SELECT a2 ...) OR t1.b1 IN (SELECT a3 ...)' and check result equivalence. Also some queries to test when subquery output is provably unique, e.g. 't1.a1 IN (SELECT a2 ... GROUP BY a2)'. Another pattern to test is 'SELECT CASE WHEN a1 IN (SELECT a2 FROM t2) THEN 1 ELSE 2 END FROM t1'.

Reply:
Thanks - that helped identify a case where I was incorrectly triggering the transformation. Fixed now - and also added test cases